src/App.tsx

```diff
@@ -3,29 +3,128 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 interface State {
+  todos: Todo[];
   selectedUserId: number;
+  isActiveTodoId: number;
+  filterQuery: string;
+  showTodos: string;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
+    todos: [],
     selectedUserId: 0,
+    isActiveTodoId: 0,
+    filterQuery: '',
+    showTodos: 'all',
+  };
+
+  componentDidMount() {
+    getTodos()
+      .then(todos => {
+        this.setState({ todos });
+      });
+  }
+
+  onRandomizer = (event: React.MouseEvent) => {
+    event.preventDefault();
+
+    this.setState(prevState => ({
+      todos: prevState.todos.sort(() => Math.random() - 0.5),
+    }));
+  };
+
+  setFilterValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      filterQuery: target.value,
+    });
+  };
+
+  setShowTodos = ({ target }: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      showTodos: target.value,
+    });
+  };
+
+  setSelectedUserId = (event: React.MouseEvent, userId: number, id: number) => {
+    event.preventDefault();
+
+    this.setState({
+      selectedUserId: userId,
+      isActiveTodoId: id,
+    });
+  };
+
+  setCompletedTodo = ({ target }: React.ChangeEvent<HTMLInputElement>, id: any) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map(todo => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: target.checked,
+          };
+        }
+
+        return todo;
+      }),
+    }));
   };
 
   render() {
-    const { selectedUserId } = this.state;
+    const {
+      todos,
+      selectedUserId,
+      isActiveTodoId,
+      filterQuery,
+      showTodos,
+    } = this.state;
+
+    let newTodos;
+
+    switch (showTodos) {
+      case 'active':
+        newTodos = todos.filter(({ completed }) => completed);
```

Comment:
```suggestion
      case 'active':
        newTodos = todos.filter(({ completed }) => !completed);
```
Мало би бути навпаки. active - значить "активне", отже "незавершене", тому мало б бути НЕ completed

Reply:
Дякую!)
Щодо active якось не зверув уваги - виправляюся)